### DISCUSSION TITLE
Template string or variable usage in dynamic imports

### DISCUSSION THREAD

**Discussion by kollerdroid** [vercel/next.js#33980] [Ideas]
### Describe the feature you'd like to request

We would like to build pages dynamically based on a (json) descriptor. There are many components available on a CDN, but for a particular page creation only some of them is required, which are declared in the descriptor.
So we want to load some (only the needed) components dynamically, avoiding to import all available components statically.

### Describe the solution you'd like

Please enable template string or variable usage in server side dynamic imports to achieve 'dynamically constructed' component loading.

Eg.:

```js
import dynamic from 'next/dynamic';

const componentName = 'Card';
const DynamicComponent = dynamic(() => import(`https://my.cdn.com/components/${componentName}.js`));

// or

const componentUrl = `https://my.cdn.com/components/${componentName}.js`;
const DynamicComponent = dynamic(() => import(componentUrl));
```

### Describe alternatives you've considered

On client side this functionality is available and in such cases is recommended.

According to [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports):

> The following are some reasons why you might need to use dynamic import:
> 
> ...
> When the import specifier string needs to be constructed dynamically.   
> ...

```js
const componentName = 'Card';

const comp = await import(`https://my.cdn.com/components/${componentName}.js`);

// or 

const componentUrl = `https://my.cdn.com/components/${componentName}.js`;
const comp = await import(componentUrl);
```

**Comment by dwbruhn** [comment]
👍 for this request! This would eliminate manual maintenance of component lists and repetitive import statements that bloat the code.

I have a use case where I need to dynamically import one of many possible components. Because all the `import` statements have to be explicit paths, the lookup file is probably going to be large, so I may even need to chunk the lookup file itself and dynamically import it in portions. 😞 

Maybe this would be a good time to reconsider [this issue](https://github.com/vercel/next.js/issues/5065), where [loadable-components](https://loadable-components.com/) was recommended to replace [the fork of react-loadable in next/dynamic](https://github.com/vercel/next.js/blob/401a9a4479dee719929a2b4bb708e93912979814/packages/next/shared/lib/loadable.js).

Loadable-components [supports variables in the import path](https://loadable-components.com/docs/dynamic-import/), but [react-loadable](https://github.com/jamiebuilds/react-loadable) apparently does not.

**Comment by cdbattags** [comment]
This is currently working for me but I get a hydration mismatch error in Next 12.3.1
